homeassistant/components/homeworks/config_flow.py

```diff
@@ -0,0 +1,465 @@
+"""Lutron Homeworks Series 4 and 8 config flow."""
+from __future__ import annotations
+
+from functools import partial
+import logging
+from typing import Any
+
+from pyhomeworks.pyhomeworks import Homeworks
+import voluptuous as vol
+
+from homeassistant.components.light import DOMAIN as LIGHT_DOMAIN
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.core import (
+    DOMAIN as HOMEASSISTANT_DOMAIN,
+    HomeAssistant,
+    async_get_hass,
+    callback,
+)
+from homeassistant.data_entry_flow import AbortFlow
+from homeassistant.helpers import (
+    config_validation as cv,
+    entity_registry as er,
+    issue_registry as ir,
+    selector,
+)
+from homeassistant.helpers.schema_config_entry_flow import (
+    SchemaCommonFlowHandler,
+    SchemaFlowError,
+    SchemaFlowFormStep,
+    SchemaFlowMenuStep,
+    SchemaOptionsFlowHandler,
+)
+from homeassistant.helpers.selector import TextSelector
+from homeassistant.util import slugify
+
+from . import DEFAULT_FADE_RATE, calculate_unique_id
+from .const import (
+    CONF_ADDR,
+    CONF_CONTROLLER_ID,
+    CONF_DIMMERS,
+    CONF_INDEX,
+    CONF_KEYPADS,
+    CONF_RATE,
+    DEFAULT_KEYPAD_NAME,
+    DEFAULT_LIGHT_NAME,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONTROLLER_EDIT = {
+    vol.Required(CONF_HOST): selector.TextSelector(),
+    vol.Required(CONF_PORT): selector.NumberSelector(
+        selector.NumberSelectorConfig(
+            min=1,
+            max=65535,
+            mode=selector.NumberSelectorMode.BOX,
+        )
+    ),
+}
+
+LIGHT_EDIT = {
+    vol.Optional(CONF_RATE, default=DEFAULT_FADE_RATE): selector.NumberSelector(
+        selector.NumberSelectorConfig(
+            min=0,
+            max=20,
+            mode=selector.NumberSelectorMode.SLIDER,
+            step=0.1,
+        )
+    ),
+}
+
+
+validate_addr = cv.matches_regex(r"\[\d\d:\d\d:\d\d:\d\d\]")
+
+
+async def validate_add_controller(
+    handler: ConfigFlow | SchemaOptionsFlowHandler, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate controller setup."""
+    user_input[CONF_CONTROLLER_ID] = slugify(user_input[CONF_NAME])
```

Comment:
Lutron Homeworks Series 4 and 8 has no serial number or other unique identifier which can be read by the client, so we use the slugified controller name to deduplicate config entries and as a base for entity unique ID. Host name could be used to deduplicate config entries, but it would not be useful as a entity ID base because it may change.